test/functional/api/window_spec.lua

```diff
@@ -322,5 +322,20 @@ describe('API/win', function()
       meths.win_close(oldwin,true)
       eq({newwin}, meths.list_wins())
     end)
+
+    it('in cmdline-window #9767', function()
+      command('split')
+      eq(2, #meths.list_wins())
+      -- Open cmdline-window.
+      feed('q:')
+      eq(3, #meths.list_wins())
+      eq(':', funcs.getcmdwintype())
+      -- Vim: not allowed to close other windows from cmdline-window.
```

Comment:
Was this intended to be forbidden? The test does not test it, `1` is never a valid window id. Also it doesn't look implemented (`cmdwin_type != 0` and then `win != curwin` needs to return error explicitly).

Reply:
Ok, then I misunderstood whatever test was in this PR before.

Reply:
It can create weird behaviors, it seems. I already did https://github.com/neovim/neovim/pull/10738 :)